lib/manager/components/deployment/FeedVersionTableRow.js

```diff
@@ -0,0 +1,274 @@
+
+import {
+  deleteFeedVersion,
+  togglePinFeedVersion,
+  updateVersionForFeedSource
+} from '../../actions/deployments'
+import {formatTimestamp, fromNow} from '../../../common/util/date-time'
+import {
+  versionHasExpired,
+  versionHasNotBegun,
+  versionIsPinnedInDeployment
+} from '../../util/version'
+
+import type {
+  Deployment,
+  Feed,
+  Project,
+  SummarizedFeedVersion
+} from '../../../types'
+import type {AppState} from '../../../types/reducers'
+
+function addClassIfFalsy (val, className) {
+  return !val ? ` ${className}` : ''
+}
+
+function warnIfFalsy (val) {
+  return addClassIfFalsy(val, 'warning')
+}
+
+function dangerIfTruthy (val) {
+  return addClassIfFalsy(!val, 'danger')
+}
+
+type RowProps = {
+  deployment: Deployment,
+  feedSource: Feed,
+  project: Project,
+  version: SummarizedFeedVersion
+}
+
+type ConnectedRowProps = RowProps & {
+  deleteFeedVersion: typeof deleteFeedVersion,
+  feedVersionPinned: boolean,
+  togglePinFeedVersion: typeof togglePinFeedVersion,
+  updateVersionForFeedSource: typeof updateVersionForFeedSource,
+  userCanDecrementVersion: boolean,
+  userCanEditDeployment: boolean,
+  userCanIncrementVersion: boolean,
+  userCanRemoveVersion: boolean
+}
+
+class FeedVersionTableRow extends Component<ConnectedRowProps> {
+  _decrementVersion = () => {
+    const {deployment, feedSource, updateVersionForFeedSource, version} = this.props
+    const {previousVersionId: id} = version
+    if (!id) {
+      console.warn('No previous version ID exists for version', version)
+      return
+    }
+    updateVersionForFeedSource(deployment, feedSource, {id})
+  }
+
+  _incrementVersion = () => {
+    const {
+      deployment,
+      feedSource,
+      updateVersionForFeedSource,
+      version
+    } = this.props
+    const {nextVersionId: id} = version
+    if (!id) {
+      console.warn('No next version ID exists for version', version)
+      return
+    }
+    updateVersionForFeedSource(deployment, feedSource, {id})
+  }
+
+  _incrementVersionToLatest = () => {
+    const {deployment, feedSource, updateVersionForFeedSource} = this.props
+    if (!feedSource.latestVersionId) {
+      console.warn('No latest version ID exists for feedSource', feedSource)
+      return
+    }
+    updateVersionForFeedSource(
+      deployment,
+      feedSource,
+      {id: feedSource.latestVersionId}
+    )
+  }
+
+  _togglePinFeedVersion = () => {
+    const {togglePinFeedVersion, deployment, version} = this.props
+    togglePinFeedVersion(deployment, version)
+  }
+
+  _removeVersion = () => {
+    const {deleteFeedVersion, deployment, version} = this.props
+    deleteFeedVersion(deployment, version)
+  }
+
+  render () {
+    const {
+      feedSource,
+      feedVersionPinned,
+      userCanDecrementVersion,
+      userCanEditDeployment,
+      userCanIncrementVersion,
+      userCanRemoveVersion,
+      version
+    } = this.props
+    const {validationResult: result} = version
```

Comment:
I think qualifying some things with where they come from is helpful here instead of just having all these single name variables that might be hard to trace back to their origin.